src/main/java/com/aws/greengrass/util/platforms/unix/UnixPlatform.java

```diff
@@ -549,17 +549,27 @@ private List<String> children(String parent, Map<String, List<String>> procMap)
         return ret;
     }
 
-    private String getIpcServerSocketAbsolutePath(Path rootPath) {
-        return rootPath.resolve(IPC_SERVER_DOMAIN_SOCKET_FILENAME).toString();
+    private String getIpcServerSocketAbsolutePath(Path rootPath, Path ipcPath) {
+        if (ipcPath == null) {
+            return rootPath.resolve(IPC_SERVER_DOMAIN_SOCKET_FILENAME).toString();
+        }
+        return ipcPath.toString();
     }
 
     private boolean isSocketPathTooLong(String socketPath) {
         return socketPath.length() >= UDS_SOCKET_PATH_MAX_LEN;
     }
 
     @Override
-    public String prepareIpcFilepath(Path rootPath) {
-        String ipcServerSocketAbsolutePath = getIpcServerSocketAbsolutePath(rootPath);
+    public String prepareIpcFilepath(Path rootPath, Path ipcPath) {
+        String ipcServerSocketAbsolutePath = getIpcServerSocketAbsolutePath(rootPath, ipcPath);
+
+        try {
+            Utils.createPaths(Paths.get(ipcServerSocketAbsolutePath));
```

Comment:
createPaths needs a directory. This path will be the path to a file, not a directory. Get the parent path first, then create the paths.